### PR TITLE
New language: C++ (Windows)

### PR DESCRIPTION
**Pull request by npanuhin**
Tested on Windows 10. Currently uses `.exe` files, so compilation can not work on Linux.

Also added settings file with commands that are used to compile and run code.
(`Preferences > Package Settings > EasyCP settings`)

### CODE REVIEW

**Comment by npanuhin** [comment]
Also, I want to say that the ability to parse all problems at once should be added (simple), and the ability to parse problems in PDF format (a bit harder).

**Comment by RohitMazumder** [comment]
In the CompileCommand class, for file_extension: java and cpp, much of the code is similar. It will be great if we can avoid the duplicate code.

**Comment by RohitMazumder** [comment]
> Also, I want to say that the ability to parse all problems at once should be added (simple), and the ability to parse problems in PDF format (a bit harder).

Parse all the problems? 
As in parse all problems in a contest?

**Comment by npanuhin** [comment]
> Parse all the problems?
> As in parse all problems in a contest?

I meant:
Now plugin can only parse one task, e.g. https://codeforces.com/contest/1284/problem/A
It would be great if it could parse, for example, https://codeforces.com/contest/1284 and create folders for all A-G problems.
Thus, the user won’t have to run it several times for one contest.
_(If you agree, I will implement this feature a bit later)_

File: easycp.py [hunk 1]
```diff
@@ -64,18 +65,33 @@ def get_num_tests():
 
         def get_output():
 
+            if file_extension == 'java':
+                cmd = settings.get("java_run", "java")
+                if type(cmd) is not list:
+                    cmd = [cmd]
+                cmd += ['java', '-cp', classpath, file_name]
```
Comment (RohitMazumder) [inline easycp.py]: `cmd += ['-cp', classpath, file_name]`

**Comment by npanuhin** [comment]
Not "hotkeys". I meant "sublime-commands"

**Comment by npanuhin** [comment]
Currently, package uses commands:
```
parse_url
compile
run
```
But I think they are too general and may overlap with other plugin's commands.
Maybe they can be replaced by these?
```
easycp_parse_url
easycp_compile
easycp_run
```

**Comment by npanuhin** [comment]
And I think a description for the package should be added, which contains the words "codeforces" and "tests". For the PackageControl's browse system.

Also, I think when this PR will be merged, labels "python", "c++" and "codeforces" should be added to package_control_channel.

**Comment by RohitMazumder** [comment]
> And I think a description for the package should be added, which contains the words "codeforces" and "tests". For the PackageControl's browse system.
> 
> Also, I think when this PR will be merged, labels "python", "c++" and "codeforces" should be added to package_control_channel.

Yes thanks !

**Comment by RohitMazumder** [comment]
> Currently, package uses commands:
> 
> ```
> parse_url
> compile
> run
> ```
> 
> But I think they are too general and may overlap with other plugin's commands.
> Maybe they can be replaced by these?
> 
> ```
> easycp_parse_url
> easycp_compile
> easycp_run
> ```

Yes, I agree.

**Comment by Nikita-Panyuhin** [comment]
Why do not you what to use bs4 (BeautifulSoup) Python library to parse HTML? I think it will be easier to use bs4, rather than html.parser (HTMLParser).
_([It is available as a Package Control dependency](https://github.com/wbond/package_control_channel/blob/master/repository/dependencies.json#L76-L90))_

**Pull request merged**